### PR TITLE
refactor: add shared types (Bruception)

### PR DESCRIPTION
**Pull request by Bruception**
### Description

Closes #

### CODE REVIEW

**Comment by Bruception** [comment]
boohoo no more ambients 
image

File: frontend/src/ts/ape/server-configuration.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import Ape from ".";
+/* eslint-disable-next-line */
```
Comment (Bruception) [inline frontend/src/ts/ape/server-configuration.ts:2]: Okay, we should figure out how to relax this rule for these special cases. So we don't have this line peppered throughout the project. I have left this as an exercise to the reviewers @Miodec @fehmer ;)

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/6798990518

**Review by Miodec — APPROVED** [review]
Thanks

**Pull request merged**